csharp/device/Microsoft.Azure.Devices.Client/Common/SHA.cs

```diff
@@ -0,0 +1,211 @@
+// 
+//  Code by Elze Kool
```

Comment:
If it's to anyone's help, I indeed mailed José the above, so it's okay. Good luck with your project!